ChatSX/components/Poll/index.tsx

```diff
@@ -0,0 +1,1689 @@
+        conversationId: item?.id,
+        poll: pollObject,
+      });
+
+      await reloadConversation();
+    } catch (error) {
+      // process error
+    }
+  }
+
+  // this function used we interact with poll options
+  async function setSelectedPollOptions(pollIndex: any) {
+    if (Date.now() > item?.expiry_time) {
+      dispatch({
+        type: SHOW_TOAST,
+        body: {isToast: true, msg: POLL_ENDED_WARNING},
+      });
+      return;
+    }
+    const newSelectedPolls = [...selectedPolls];
+    const isPollIndexIncluded = newSelectedPolls.includes(pollIndex);
+
+    if (isPollIndexIncluded) {
+      // if poll item is already selected
+      const isSelected = item?.polls?.some((poll: any) => {
+        return poll?.is_selected;
+      });
+      // if (isSelected) {
+      //   alert('Already voted');
+      //   return;
+      // }
+      const selectedIndex = newSelectedPolls.findIndex(
+        index => index === pollIndex,
+      );
+      newSelectedPolls.splice(selectedIndex, 1);
+    } else {
+      const isSelected = pollsArr?.some((poll: any) => {
+        return poll?.is_selected;
+      });
+
+      // Already submitted poll condition
+      if (isSelected && item?.poll_type === 0) {
+        return;
+      } else if (item?.poll_type === 1 && shouldShowVotes) {
+        return;
+      }
+
+      // if only one option is allowed
+      if (item?.multiple_select_no === 1) {
+        // can change selected ouptput in deferred poll
+        if (item?.poll_type === 1) {
+          const pollSubmissionCall = await myClient.submitPoll({
+            conversationId: item?.id,
+            polls: [item?.polls[pollIndex]],
+          });
+          await reloadConversation();
+          dispatch({
+            type: SHOW_TOAST,
+            body: {isToast: true, msg: POLL_SUBMITTED_SUCCESSFULLY},
+          });
+        } else {
+          // for instant poll selection only for once
+
+          // if not selected
+          if (!isSelected) {
+            const pollSubmissionCall = await myClient.submitPoll({
+              conversationId: item?.id,
+              polls: [item?.polls[pollIndex]],
+            });
+            await reloadConversation();
+            dispatch({
+              type: SHOW_TOAST,
+              body: {isToast: true, msg: POLL_SUBMITTED_SUCCESSFULLY},
+            });
+          }
+        }
+        return;
+      }
+
+      // multiple options are allowed
+      switch (item?.multiple_select_state) {
+        case 0: {
+          if (selectedPolls.length === item?.multiple_select_no) {
+            return;
+          }
+          break;
+        }
+        case 1: {
+          if (selectedPolls.length == item?.multiple_select_no) {
+            return;
+          }
+          break;
+        }
+      }
+      newSelectedPolls.push(pollIndex);
+    }
+    setSelectedPolls(newSelectedPolls);
+  }
+
+  // this function call submit poll button API
+  async function submitPoll() {
```

Comment:
why we are making API call in UI class, by this way, will not be able to segregate UI and Business logic